### PR TITLE
add endpoint to list Vsphere datastores

### LINKED ISSUE #6356: Create new endpoint to list VMware Datasore
**User Story**

**Acceptance criteria**

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: add an endpoint to list Vsphere datastores. As a result, the endpoint returns the following list:
```
{
  "datastores": [
    "/dc-1/datastore/esxi-3",
    "/dc-1/datastore/esxi-2",
    "/dc-1/datastore/esxi-4",
    "/dc-1/datastore/esxi-5",
    "/dc-1/datastore/esxi-1",
    "/dc-1/datastore/dsc-1/exsi-nas"
  ]
}
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6356

```release-note
add an endpoint to list Vsphere datastores: GET /api/v2/providers/vsphere/datastores
```

### CODE REVIEW

File: cmd/kubermatic-api/swagger.json [hunk 1]
```diff
@@ -12984,6 +12984,57 @@
...
+            "in": "header"
+          },
+          {
+            "type": "string",
+            "name": "Password",
+            "in": "header"
+          },
+          {
+            "type": "string",
+            "name": "DatacenterName",
```
Comment (irozzo-1A) [inline cmd/kubermatic-api/swagger.json:13010]: nit: personally I think that using a path param would be more appropriate here

**Review by irozzo-1A — COMMENTED** [review]
LGTM, just a comment about the design of the API

File: cmd/kubermatic-api/swagger.json (hunk 1 above)
Comment (zreigz) [reply cmd/kubermatic-api/swagger.json:13010]: this is what we have also for other provider endpoints so want to keep this consistent

Comment (irozzo-1A) [reply cmd/kubermatic-api/swagger.json:13010]: ok, no pb it's not a blocker.

**Pull request merged**